odoo/cli/shell.py

```diff
@@ -80,27 +83,34 @@ def console(self, local_vars):
 
             for shell in shells_to_try:
                 try:
-                    return getattr(self, shell)(local_vars)
+                    return getattr(self, shell)(local_vars, pythonstartup)
                 except ImportError:
                     pass
                 except Exception:
                     _logger.warning("Could not start '%s' shell." % shell)
                     _logger.debug("Shell error:", exc_info=True)
 
-    def ipython(self, local_vars):
+    def ipython(self, local_vars, pythonstartup=None):
         from IPython import start_ipython
-        start_ipython(argv=[], user_ns=local_vars)
+        start_ipython(argv=[
+            "--TerminalIPythonApp.display_banner=False",
+            f"--TerminalIPythonApp.exec_files={pythonstartup}"
```

Comment:
@kmagusiak Since we are executing this file, I checked it to be a valid path to avoid  injections... Do you think it's redundant, since we're opening a Python shell with the Odoo user's permissions?

Reply:
What do you mean by checking the path? Inject the path just if it was provided.

Reply:
Looks like the various shell already do the check 
![immagine](https://github.com/user-attachments/assets/f371b2e9-ff41-44c4-819e-cc2532d4116a)

Reply:
Yes, just pass the arguments and the shells will handle them as usual.